privacyidea/lib/crypto.py

```diff
@@ -656,10 +665,10 @@ def sign(self, s):
             s = s.encode('utf8')
         RSAkey = RSA.importKey(self.private)
```

Comment:
Shouldn't we also replace the RSA signing and verification? 
See https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#signing
and https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#verification

However, we should use pkcs1, as we did before:

Verify:
~~~~python
pubkey_obj = serialization.load_pem_public_key(str(pubkey_pem), default_backend())
chosen_hash = hashes.SHA256()
hasher = hashes.Hash(chosen_hash, default_backend())
sign_data = u"{0!s}|{1!s}".format(challenge, serial)
hasher.update(sign_data.encode("utf8"))
digest = hasher.finalize()
try:
                        pubkey_obj.verify(b32decode(signature), digest,
                                          padding.PKCS1v15(),
                                          utils.Prehashed(chosen_hash))
~~~~

We use this in the new pushtoken. 
Should I refactor this Sign Object, so that it can be used in the push token?

Reply:
This will be addressed in Part 2 :-)